zebra-state/src/request.rs

```diff
@@ -257,41 +257,54 @@ impl Treestate {
 /// database and recompute a new one.
 pub struct ContextuallyVerifiedBlockWithTrees {
     /// A block ready to be committed.
-    pub block: SemanticallyVerifiedBlock,
+    pub verified: SemanticallyVerifiedBlock,
     /// The tresstate associated with the block.
-    pub treestate: Option<Treestate>,
+    pub treestate: Treestate,
 }
 
-impl ContextuallyVerifiedBlockWithTrees {
+pub enum FinalizableBlock {
+    Checkpoint {
+        checkpoint_verified: CheckpointVerifiedBlock,
+    },
+    Contextual {
+        contextually_verified: ContextuallyVerifiedBlock,
+        treestate: Treestate,
+    },
+}
+
+impl FinalizableBlock {
     pub fn new(contextually_verified: ContextuallyVerifiedBlock, treestate: Treestate) -> Self {
```

Comment:
Can you add documentation to these methods? It can just be short, but it helps to explain what they do.

Reply:
Added in https://github.com/ZcashFoundation/zebra/pull/7035/commits/d61add359e14cf2251b3c76053a7653e09607314

Reply:
Refactored in [55def20](https://github.com/ZcashFoundation/zebra/pull/7035/commits/55def2032ae6b070251dff7ab8cce8ab9404228d).